packages/at_client/lib/src/transformer/response_transformer/notification_response_transformer.dart

```diff
@@ -76,6 +77,12 @@ class NotificationResponseTransformer
     return atNotification;
   }
 
+  // Decrypt the notification key only if isEncrypted flag is set to true or
+  // null (for backward compatibility - default behavior is decrypt the notification key)
+  bool _shouldDecryptKey(bool? isEncrypted) {
```

Comment:
I think the doc should be like "Returns true when a notification is encrypted and needs a decryption etc.." also prefer this method taking a notification as an argument

Reply:
Then we can write tests around that flag being null, true and false

Reply:
> I think the doc should be like "Returns true when a notification is encrypted and needs a decryption etc.." also prefer this method taking a notification as an argument

there are two fields on which we do encryption/decryption. 1) notification key 2) notification value. hence i didn't add a method to decrypt by passing in notification object

Reply:
Not to decrypt, it is the bool method.